pkg/operator/revisioncontroller/revision_controller.go

```diff
@@ -353,42 +353,59 @@ func (c RevisionController) sync(ctx context.Context, syncCtx factory.SyncContex
 		return nil
 	}
 
-	if shouldCreateNewRevision, err := c.revisionPrecondition(ctx); err != nil || !shouldCreateNewRevision {
-		return err
-	}
-
 	// If the operator status's latest available revision is not the same as the observed latest revision, update the operator.
+	// This needs to be done even if the revision precondition is not required because it ensures our operator status is
+	// correct for all consumers.
+	// This is what is going to allow us to move where the state is stored in authentications.openshift.io.
 	latestObservedRevision, err := c.getLatestAvailableRevision(ctx)
 	if err != nil {
 		return err
 	}
-	if latestObservedRevision != 0 && latestAvailableRevisionSeenByOperator != latestObservedRevision {
+	if latestObservedRevision != 0 && operatorStatus.LatestAvailableRevision != latestObservedRevision {
 		// Then make sure that revision number is what's in the operator status
-		_, _, err := c.operatorClient.UpdateLatestRevisionOperatorStatus(ctx, latestObservedRevision)
+		status := applyoperatorv1.OperatorStatus().
+			WithConditions(applyoperatorv1.OperatorCondition().
+				WithType(condition.RevisionControllerDegradedConditionType).
+				WithStatus(operatorv1.ConditionFalse),
+			).
+			WithLatestAvailableRevision(latestObservedRevision)
+		err := c.operatorClient.ApplyOperatorStatus(ctx, c.controllerInstanceName, status)
 		if err != nil {
 			return err
 		}
 		// regardless of whether we made a change, requeue to rerun the sync with updated status
 		return factory.SyntheticRequeueError
 	}
 
-	requeue, syncErr := c.createRevisionIfNeeded(ctx, syncCtx.Recorder(), latestAvailableRevisionSeenByOperator, resourceVersion)
+	if shouldCreateNewRevision, err := c.revisionPrecondition(ctx); err != nil || !shouldCreateNewRevision {
+		return err
+	}
+
+	wroteStatus, requeue, syncErr := c.createRevisionIfNeeded(ctx, syncCtx.Recorder(), operatorStatus.LatestAvailableRevision)
+	switch {
+	case requeue && syncErr == nil:
+		return factory.SyntheticRequeueError
+	case syncErr != nil:
+		// this is updated in status inside of createRevisionIfNeeded
+		return syncErr
+
+	}
 	if requeue && syncErr == nil {
 		return factory.SyntheticRequeueError
 	}
-	err = syncErr
+	if wroteStatus {
+		return syncErr
+	}
 
 	// update failing condition
-	cond := operatorv1.OperatorCondition{
-		Type:   condition.RevisionControllerDegradedConditionType,
-		Status: operatorv1.ConditionFalse,
-	}
-	if err != nil {
-		cond.Status = operatorv1.ConditionTrue
-		cond.Reason = "Error"
-		cond.Message = err.Error()
-	}
-	if _, _, updateError := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(cond)); updateError != nil {
+	status := applyoperatorv1.OperatorStatus().
```

Comment:
if the `wroteStatus == false` then we should add the `syncErr` msg.

Reply:
>if the wroteStatus == false then we should add the syncErr msg.

At this point `syncErr` is guaranteed to be nil, so there's nothing to include